### PR TITLE
Calculate and save off match census data once a day

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/match_census.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+###
+# Copyright 2016 - 2019 Green River Data Analysis, LLC
```
Comment (awisema) [inline app/models/match_census.rb]: 2020 :-)

File: app/models/match_census.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#
+# License detail: https://github.com/greenriver/boston-cas/blob/master/LICENSE.md
+###
+
+class MatchCensus < ActiveRecord::Base
+  belongs_to :opportunity
+  belongs_to :match, class_name: 'ClientOpportunityMatch', required: false
+
+  def self.populate!
+    # Add all opportunities that have not been ma
```
Comment (awisema) [inline app/models/match_census.rb]: ?

File: app/models/match_census.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      populate_from_opportunity!(opp)
+    end
+  end
+
+  def self.populate_from_opportunity! opp
+    self.transaction do
+      # Clear any existing data for this opportunity on this day, and rebuild
+      where(opportunity_id: opp.id, date: Date.current).delete_all
+      clients_for_route = Client.available_for_matching(opp.match_route)
+      available_client_ids = opp.matching_clients(clients_for_route).pluck(:id)
```
Comment (awisema) [inline app/models/match_census.rb]: Add comment noting that this is sorted by the prioritization function on the match route?

File: app/models/match_census.rb (hunk 1 above)
Comment (eanders) [reply app/models/match_census.rb]: Yep, that's my next PR

**Pull request merged**